airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -15,17 +15,26 @@
 # specific language governing permissions and limitations
 # under the License.
 """Executes task in a Kubernetes POD"""
+import json
 import re
 import warnings
-from typing import TYPE_CHECKING, Any, Dict, Iterable, List, Optional, Tuple, Type
+from contextlib import AbstractContextManager
+from typing import TYPE_CHECKING, Any, Dict, Iterable, List, Optional
 
 from kubernetes.client import CoreV1Api, models as k8s
 
+from airflow.providers.cncf.kubernetes.utils.pod_launcher import PodLaunchFailedException, PodStatus
+
 try:
     import airflow.utils.yaml as yaml
 except ImportError:
     import yaml
 
+try:
+    from functools import cached_property
+except ImportError:
+    from cached_property import cached_property
+
```

Comment:
```suggestion
from airflow.compat.functools import cached_property

```

Reply:
naah we can't do that.. since this is provider change and we want to avoid breaking old Airflow's that don't have `airflow.compat` module we will have to use `try...except`